Revamp Inference Providers doc (#1652)
huggingface/hub-docs: docs/api-inference/pricing.md

> # Pricing and Billing

Reviewer: 👍

> Inference Providers is a production-ready service involving external partners and is therefore a paid-product. However, as a Hugging Face user you get monthly credits to run experiments. The amount of credits you get depends on your type of account:

Reviewer 2: ```suggestion
Inference Providers is a production-ready service involving external partners and is therefore a paid product. However, as a Hugging Face user, you get monthly credits to run experiments. The amount of credits you get depends on your type of account:
```

> The Inference Providers is a production-ready service involving external partners and is therefore a paid-product. However, as an Hugging Face user you get monthly credits to run experiments. The amount of credits you get depends on your type of account:

Reviewer 3: ```suggestion
Inference Providers is a production-ready service involving external partners and is therefore a paid-product. However, as a Hugging Face user you get monthly credits to run experiments. The amount of credits you get depends on your type of account:
```

| User Tier                | Included monthly credits           |
| ------------------------ | ---------------------------------- |
| Free Users               | subject to change, less than $0.10 |

> | PRO and Enterprise Users | $2.00                              |

Reviewer 4: it would be great to have a third column with an estimation on how much I can generate (might be hard)

## Pay-as-you-Go

> Only **PRO and Enterprise Hub users** can continue using the API once their monthly included credits are exhausted. This billing model, known as "Pay-as-you-Go" (PAYG), is charged on top of the monthly subscription. PAYG is only available for providers that are integrated with our billing system. We're actively working to integrate all providers, but in the meantime, any providers that are not yet integrated will be blocked once the free-tier limit is reached.

Reviewer: ```suggestion
**PRO and Enterprise Hub users** can continue using the API once their monthly included credits are exhausted. This billing model, known as "Pay-as-you-Go" (PAYG), is charged on top of the monthly subscription. PAYG is only available for providers that are integrated with our billing system. We're actively working to integrate all providers, but in the meantime, any providers that are not yet integrated will be blocked once the free-tier limit is reached.
```

> If you haven't used up your included credits yet, we estimate costs for providers that aren’t fully integrated with our billing system. These estimates are usually higher than the actual cost to prevent abuse, which is why PAYG is currently disabled for those providers.

Reviewer 3: ```suggestion
If you have remaining credits, we estimate costs for providers that aren’t fully integrated with our billing system. These estimates are usually higher than the actual cost to prevent abuse, which is why PAYG is currently disabled for those providers.
```

You can track your spending on your [billing page](https://huggingface.co/settings/billing).

<Tip>

Hugging Face charges you the same rates as the provider, with no additional fees.

</Tip>

## Routed requests vs direct calls

The documentation above assumes you are making routed requests to external providers. In practice, there are 3 different ways to run inference, each with unique billing implications:

> - **Routed Request**: This is the default method for using the Inference Providers. Simply use the JavaScript or Python `InferenceClient`, or make raw HTTP requests with your Hugging Face User Access Token. Your request is automatically routed through Hugging Face to the provider's platform. No separate provider account is required, and billing is managed directly by Hugging Face. This approach lets you seamlessly switch between providers without additional setup.

Reviewer 3: ```suggestion
- **Routed Request**: This is the default method for using Inference Providers. Simply use the JavaScript or Python `InferenceClient`, or make raw HTTP requests with your Hugging Face User Access Token. Your request is automatically routed through Hugging Face to the provider's platform. No separate provider account is required, and billing is managed directly by Hugging Face. This approach lets you seamlessly switch between providers without additional setup.
```

Here is a table that sums up what we've seen so far:

|                                    | HF routing | Billed by    | Free-tier included | Pay-as-you-go                                   | Integration                               |
| ---------------------------------- | ---------- | ------------ | ------------------ | ----------------------------------------------- | ----------------------------------------- |
| **Routed request**                 | Yes        | Hugging Face | Yes                | Only for PRO users and for integrated providers | SDKs, Playground, widgets, Data AI Studio |
| **Routed request with custom key** | Yes        | Provider     | No                 | Yes                                             | SDKs, Playground, widgets, Data AI Studio |
| **Direct call**                    | No         | Provider     | No                 | Yes                                             | SDKs only                                 |

> ## HF Inference cost

Reviewer: ```suggestion
## HF-Inference cost
```

> As you may have noticed, you can select to work with `"hf-inference"` provider. This is what used to be the "Inference API (serverless)" prior to the Inference Providers integration. From a user point of view, working with HF Inference is the same as with any other providers. Past the free-tier credits, you get charged for every inference request based on the compute time x price of the underlying hardware.

Reviewer 3: ```suggestion
As you may have noticed, you can select to work with `"hf-inference"` provider. This service used to be "Inference API (serverless)" prior to Inference Providers. From a user point of view, working with HF Inference is the same as with any other provider. Past the free-tier credits, you get charged for every inference request based on the compute time x price of the underlying hardware.
```

For instance, a request to [black-forest-labs/FLUX.1-dev](https://huggingface.co/black-forest-labs/FLUX.1-dev) that takes 10 seconds to complete on a GPU machine that costs $0.00012 per second to run, will be billed $0.0012.

> The `"hf-inference"` provider is the default provider when working with the JavaScript and Python SDKs.

Reviewer: ```suggestion
The `"hf-inference"` provider is currently the default provider when working with the JavaScript and Python SDKs. Note that this default might change in the future.
```

## Organization billing

> For Enterprise Hub organizations, it is possible to centralize billing for all your users. Each user still use their own User Access Token but the requests are billed to your organization. This can be done by passing `"X-HF-Bill-To: my-org-name"` as header in your HTTP requests.

Reviewer 3: ```suggestion
For Enterprise Hub organizations, it is possible to centralize billing for all your users. Each user still uses their own User Access Token but the requests are billed to your organization. This can be done by passing `"X-HF-Bill-To: my-org-name"` as header in your HTTP requests.
```
